server/src/test/scala/org/http4s/server/middleware/authentication/AuthMiddlewareSpec.scala

```diff
@@ -76,6 +77,90 @@ class AuthMiddlewareSpec extends Http4sSpec {
       service.orNotFound(Request[IO](method = Method.POST)) must returnStatus(Ok)
       service.orNotFound(Request[IO](method = Method.GET)) must returnStatus(NotFound)
     }
+
+    "return 404 for an unmatched but authenticated route" in {
+      val userId: User = 42
+
+      val authUser: Kleisli[OptionT[IO, ?], Request[IO], User] =
+        Kleisli.pure(userId)
+
+      val authedService: AuthedService[User, IO] =
+        AuthedService {
+          case POST -> Root as _ => Ok()
+        }
+
+      val middleware = AuthMiddleware(authUser)
+
+      val service = middleware(authedService)
+
+      service.orNotFound(Request[IO](method = Method.POST)) must returnStatus(Ok)
+      service.orNotFound(Request[IO](method = Method.GET)) must returnStatus(NotFound)
+    }
+
+    "return 401 for a matched, but unauthenticated route" in {
+      val authUser: Kleisli[OptionT[IO, ?], Request[IO], User] =
+        Kleisli.lift(OptionT.none)
+
+      val authedService: AuthedService[User, IO] =
+        AuthedService {
+          case POST -> Root as _ => Ok()
+        }
+
+      val middleware = AuthMiddleware(authUser)
+
+      val service = middleware(authedService)
+
+      service.orNotFound(Request[IO](method = Method.POST)) must returnStatus(Unauthorized)
+      service.orNotFound(Request[IO](method = Method.GET)) must returnStatus(Unauthorized)
```

Comment:
This is an unmatched authenticated route.  The condition is right, but maybe that's a separate test.

Reply:
I can move to a separate test for this one, sure